### PR TITLE
[explorer] Displays Transaction Timestamps

### PR DESCRIPTION
**Pull request by Andrew47**
Implements the frontend component of issue https://github.com/MystenLabs/sui/issues/1696. 

This means timestamps in the following (all snapshots using live data).

### Home Page
image

### Object Results Page
image

### Address Results Page
image

### Transaction Results Page
image

### CODE REVIEW

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 1]
```diff
@@ -114,6 +116,7 @@ function LatestTxView({
                         )}
                     >
                         <div className={styles.txcardgridlarge}>TxId</div>
+                        <div className={styles.txage}>Age</div>
```
Comment (stella3d) [inline explorer/client/src/components/transaction-card/RecentTxCard.tsx:119]: Can we label this "Time" or "Timestamp" instead (and change the associated variables)?
pictured: etherscan's version
image

File: explorer/client/src/components/transactions-for-id/TxForID.tsx [hunk 2]
```diff
@@ -58,6 +60,7 @@ function TxForIDView({ showData }: { showData: TxnData[] | undefined }) {
         <div id="tx" className={styles.txresults}>
             <div className={styles.txheader}>
                 <div className={styles.txid}>TxId</div>
+                <div className={styles.txage}>Age</div>
```
Comment (stella3d) [inline explorer/client/src/components/transactions-for-id/TxForID.tsx:63]: also apply rename here

File: explorer/client/src/utils/timeUtils.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    return `${stdToN(date.getUTCDate(), 2)} ${
+        MONTHS[date.getUTCMonth()]
+    } ${date.getUTCFullYear()} ${stdToN(date.getUTCHours(), 2)}:${stdToN(
+        date.getUTCMinutes(),
+        2
+    )}:${stdToN(date.getUTCSeconds(), 2)} UTC`;
+};
+
+export const timeAgo = (
```
Comment (stella3d) [inline explorer/client/src/utils/timeUtils.ts:37]: can we switch this out for a library like [Luxon](https://github.com/moment/luxon) ?  
this works ok for our usual time format, but using a library means we benefit from handling locale differences & extensive testing.

**Review by stella3d — CHANGES_REQUESTED** [review]
This looks good. the main thing i think we should change is using a library for time formatting.

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx (hunk 1 above)
Comment (Andrew47) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:119]: Etherscan uses the term 'Age' when aggregating Transaction Results (https://etherscan.io/txs) and the term 'Timestamp' when it comes to a Transaction Results Page (above). I am ambivalent regarding which labels we use for the time.

@bwann52 --> How should we label the time field across the website?

**Comment by Andrew47** [comment]
The PR is ready for a second review:
* The labels in all the above screenshots now read as 'Time'.
* Time fields will no longer be displayed if there is no timestamp data.

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx (hunk 1 above)
Comment (Andrew47) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx]: Following feedback on Slack from @bwann52 all labels for the timestamp field have been changed to `Time`

File: explorer/client/src/components/transactions-for-id/TxForID.tsx (hunk 2 above)
Comment (Andrew47) [reply explorer/client/src/components/transactions-for-id/TxForID.tsx]: See above - new labels implemented

**Review by 666lcz — APPROVED** [review]

**Comment by Jibz1** [comment]
@Andrew47 is there the option to show either the minute hour/s or days to account for instances where we will not display the entire time layout.
Screen Shot 2022-07-01 at 11 59 50 AM

**Comment by Andrew47** [comment]
@Jibz1 - yes

**Comment by Andrew47** [comment]
I've rebased this a second time on main and resolved the conflicts. This will need an approval from @stella3d before it can be merged onto main

File: explorer/client/src/utils/timeUtils.ts (hunk 3 above)
Comment (stella3d) [reply explorer/client/src/utils/timeUtils.ts:37]: it's not for more advanced functionality that i propose using a library for this - it's for things that mostly vary by user - time zone and locale / number formatting.

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 4]
```diff
@@ -134,6 +139,11 @@ function LatestTxView({
                                     alttext={truncate(tx.txId, TRUNCATE_LENGTH)}
                                 />
                             </div>
+                            {tx.timestamp_ms && (
```
Comment (stella3d) [inline explorer/client/src/components/transaction-card/RecentTxCard.tsx:142]: can we unify this with the copy of it in `TXForId` ?  i think realistically we will want to repeat timestamp display in quite a few places and we want to keep them consistent.

**Review by stella3d — APPROVED** [review]
I un-resolved the issue of using a library for formatting time to make sure the reasoning was understood, but it shouldn't block this for now.  I still think the shortcomings of rolling our own time formatting in a global world are significant.

The issue of unifying formatting of timestamp elements, i would also like to see done, but it can be in a follow-on. or cleanup PR.

**Pull request merged**